src/bindings/js/node/src/core_wrap.cpp

```diff
@@ -33,55 +34,37 @@ Napi::Object CoreWrap::Init(Napi::Env env, Napi::Object exports) {
 }
 
 Napi::Value CoreWrap::read_model_sync(const Napi::CallbackInfo& info) {
-    if (info.Length() == 1 && info[0].IsString()) {
-        std::string model_path = info[0].ToString();
-        std::shared_ptr<ov::Model> model = _core.read_model(model_path);
-        return ModelWrap::Wrap(info.Env(), model);
-    } else if (info.Length() != 2) {
-        reportError(info.Env(), "Invalid number of arguments -> " + std::to_string(info.Length()));
-        return Napi::Value();
-    } else if (info[0].IsString() && info[1].IsString()) {
-        std::string model_path = info[0].ToString();
-        std::string bin_path = info[1].ToString();
-        std::shared_ptr<ov::Model> model = _core.read_model(model_path, bin_path);
-        return ModelWrap::Wrap(info.Env(), model);
-    } else {
-        reportError(info.Env(), "Error while reading model.");
-        return Napi::Value();
+    ReadModelArgs* args;
+
+    try {
+        args = new ReadModelArgs(info);
+    } catch(std::runtime_error& err) {
+        reportError(info.Env(), err.what());
+
+        return info.Env().Undefined();
     }
+
+    std::shared_ptr<ov::Model> model;
+
+    if (args->model_str.empty())
+        model = _core.read_model(args->model_path, args->bin_path);
+    else
+        model = _core.read_model(args->model_str, args->weight_tensor);
+
+    return ModelWrap::Wrap(info.Env(), model);
 }
 
 Napi::Value CoreWrap::read_model_async(const Napi::CallbackInfo& info) {
-    std::string model_path;
-    std::string bin_path;
-
     try {
-        switch (info.Length()) {
-        case 2:
-            if (!info[1].IsString())
-                throw std::runtime_error("Second argument should be string");
-
-            bin_path = info[1].ToString();
-            [[fallthrough]];
-        case 1:
-            if (!info[0].IsString())
-                throw std::runtime_error("First argument should be string");
-
-            model_path = info[0].ToString();
-            break;
-
-        default:
-            throw std::runtime_error("Invalid number of arguments -> " + std::to_string(info.Length()));
-        }
-
-        ReaderWorker* _readerWorker = new ReaderWorker(info.Env(), model_path, bin_path);
+        ReadModelArgs* args = new ReadModelArgs(info);
```

Comment:
`ReadModelArgs* args = new ReadModelArgs(info);`
Where args deleted  and free memory?

Reply:
Thx, fixed